venter/views.py

```diff
@@ -77,17 +79,33 @@ def create(cls, request):
             Complaints.objects.get(id=complaint.id)
         ).data, status=201)
 
-    def update(self, request, pk):
+    @login_required_ajax
+    def upVote(self, request, pk):
+        """UpVote or unUpVote a complaint {?action}=0,1"""
+
         complaint = self.get_complaint(pk)
-        complaint.users_up_voted.add(self.request.user.profile)
+
+        value = request.GET.get("action")
+        if value is None:
+            return Response({"message": "{?action} is required"}, status=400)
+
+            # Check possible actions
+        if value == "0":
+            complaint.users_up_voted.remove(self.request.user.profile)
+        elif value == "1":
+            complaint.users_up_voted.add(self.request.user.profile)
+        else:
+            return Response({"message": "Invalid Action"}, status=400)
+
         return Response(ComplaintSerializer(
             Complaints.objects.get(id=complaint.id)
-        ).data)
+        ).data, status=204)
```

Comment:
`204` should not have a body. Either remove the body or change the status code to `200`. This is plain _wrong_ the way it is right now.